src/rust/engine/options/src/lib.rs

```diff
@@ -200,6 +200,23 @@ pub struct OptionValue<T> {
     pub value: T,
 }
 
+#[derive(Debug)]
+pub struct OptionalOptionValue<T> {
+    pub derivation: Option<Vec<(Source, T)>>,
+    pub source: Source,
+    pub value: Option<T>,
```

Comment:
It looks like this is _almost_ identical to `OptionValue<Option<T>>` (except `derivation` becomes `Option<Vec<(Source, Option<T>)>`); what do you think about using that instead?

I think we can still add this convenience method, as I believe this should work:

```rust
impl<T> OptionValue<Option<T>> {
    fn unwrap(self) -> OptionValue<T> { ... }
}
```

Alternatively, add combinator methods like `fn map<U>(self, f: FnOnce(T) -> U) -> OptionValue<U>` to `OptionValue<T>`, then the places that need `unwrap` can write `some_option_value.map(|v| v.unwrap())` or `some_option_value.map(Option::unwrap)`.

Reply:
We could. That type for a derivation would be substantial overkill, since only defaults can be None. But I'm not sure having both types is a big problem? They are lightweight containers with no associated logic.

Reply:
Thanks for the review. Will merge as-is for now, and consider simplifying this once the Python bindings are done and we can see how it all fits together.

Reply:
Yep, not a problem at all, just exploring the solution space. Happy as is!